sale_substate/models/sale_order.py

```diff
@@ -13,5 +13,5 @@ class BaseSubstateType(models.Model):
 
 
 class SaleOrder(models.Model):
-    _inherit = ["sale.order", "base.substate.mixin"]
+    _inherit = ["base.substate.mixin", "sale.order"]  # order matters!!!
```

Comment:
I understand that finding out about something as subtle deserves the appropriate amount of exclamation marks in the heat of the moment, but for future generations of developers who will maintain this code the message will still be clear without them :smile: 

```suggestion
    _inherit = ["base.substate.mixin", "sale.order"]  # order matters
```

Reply:
I think that in previous python versions, you would get the function non-deterministically, unless there are other conditions that dictate which function is loaded in which order. But it's more that it's not the first time and people seem quite unaware of the problem (if you want to do worse, do that with product.product and product.template!)

I'll do an absorb rebase when removing the temporary dependency, change is in a fixup commit.